### ISSUE TITLE
Reward edit flow can get Gold out of sync

### ISSUE THREAD

**Issue by SabreCat** [HabitRPG/habitica#3481]
If you open a Reward for editing, change the value, and purchase it while it's still open, the Gold value deducted may not stick--at next sync, your cash will change.

I run into this sometimes because I have 0-cost Rewards that are "currency exchanges": 10 Gold per $1 for an entertainment purchase, that kind of thing. So I'll edit the Reward, change the value to the total I'm paying, click it, reset the value to 0, and close. But it turns out that doesn't work. At next action, I'll be refunded the Gold I paid, because apparently what went through to the server was the 0 value.

It's an easy workaround (need to save and close the Reward before purchasing), but it is a bit of an annoyance. If it's a head-scratcher on the back end, we could fix it through UI by disabling purchase of a Reward that's still open for editing.

**Comment by deilann** [comment]
I'd actually like to disable the clicking of Habit buttons whilst editing
too.

The number of times I've clicked a habit when I meant to remove one of the
buttons...

**Comment by Alys** [comment]
"_The number of times I've clicked a habit when I meant to remove one of the buttons_" - I do that too!! And I'll accidentally buy a reward when all I want to do is edit its gold value. I blame my brain. But more importantly:

"_need to save and close the Reward before purchasing_" - This is the same as you need to do when editing a task. For example, if you have a to-do open for editing, and you change the difficulty, you need to save it before checking off the to-do otherwise you're rewarded for the original difficulty. (It initially looks as if you're rewarded for the new difficulty, but after a server sync, the reward is adjusted, exactly as you describe for rewards). Similarly (sort of), if you change a category, you need to save the task otherwise the category change will revert on server sync.

Basically, almost all edit actions are not auto-save, even though they look like they are. I think the only exception is adding or deleting checklist items. The easy way to tell of course is if the blue sync bar shoots across the top of the screen when you make a change, then it's auto-save. I think it would be nice if everything was auto-save, but that adds to the load on the server.

"_fix it through UI by disabling purchase of a Reward that's still open for editing_"  -- and also disable the ability to check off a task (or plus/minus a habit) when they're open for editing. (And as a side-effect, it would fix the stupid-brain issue I mentioned.)
